projects/portal/src/app/pages/faucet/faucet.usecase.service.ts

```diff
@@ -0,0 +1,70 @@
+import { Injectable } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import { Config, ConfigService } from 'projects/portal/src/app/models/config.service';
+import { BehaviorSubject, combineLatest, Observable } from 'rxjs';
+import { first, map } from 'rxjs/operators';
+
+@Injectable({ providedIn: 'root' })
+export class FaucetUseCaseService {
+
+  //private config$: Observable<Config | undefined>;
+
+  constructor(
+    private readonly route: ActivatedRoute,
+    private configService: ConfigService,
+  ) {
+    //this.config$ = this.configService.config$;
+    //this.denom$ = new BehaviorSubject('');
+  }
+
+  get config$(): Observable<Config | undefined> {
+    return this.configService.config$;
+  }
+
+
+  get denoms$(): Observable<string[] | undefined> {
+    return this.config$.pipe(
+      map((config) => config?.extension?.faucet?.map((faucet) => faucet.denom)),
+    );
+  }
+
+  get address$(): Observable<string> {
+    return this.route.queryParams.pipe(map((queryParams) => queryParams.address));
+  }
+
+  get amount$(): Observable<number> {
+    return this.route.queryParams.pipe(map((queryParams) => queryParams.amount));
+  }
+
+  get denom$(): BehaviorSubject<string> {
+    this.route.queryParams
+      .pipe(first())
+      .toPromise()
```

Comment:
Use `firstValueFrom` instead of `toPromise

Reply:
I stopped using behaviorSubject, keeping information in query parameters like txs.
(An error occurred when using behaviorSubject in service.)